pkg/utils/vmi.go

```diff
@@ -47,5 +52,36 @@ func (v *VmiGetter) WhoUseNad(nad *nadv1.NetworkAttachmentDefinition, nodesFilte
 		}
 	}
 
-	return afterFilter, nil
+	vmiNameList := make([]string, 0)
+	if len(afterFilter) > 0 {
+		for _, vmi := range afterFilter {
+			vmiNameList = append(vmiNameList, vmi.Namespace+"/"+vmi.Name)
+		}
+	}
+
+	vms, err := v.VMCache.GetByIndex(indexeres.VMByNetworkIndex, networkName)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, vm := range vms {
+		if vm.Namespace != nad.Namespace {
+			continue
+		}
```

Comment:
is this check needed, as networkName contains both "Namespace + name" and vms are retrieved by matching both.